### PR TITLE
Themes: Fix theme showcase banners

### LINKED ISSUE #32739: Themes: can't close the banner ad
@mendezcode Can you please fix this up when you get a chance?

#### Steps to reproduce
1. Starting at URL: https://wordpress.com/themes/ 
2. You'll see one of several banner ads.
3. Try to close the ad at the top for a theme

#### What I expected
I saw a visual close button in the corner of the ad and expected to be able to dismiss it

#### What happened instead
I was directed to the theme. This made me feel tricked.

#### Browser / OS version
Chrome - latest / MacOS - latest

#### Screenshot / Video
image

#### Context / Source

#dogfooding

### PR DESCRIPTION
**Pull request by crunnells**
This PR fixes a few things related to the theme showcase banner code. Mostly fixes an issue where the close button wasn't accessible. There's also a visual problem, shown here:

Screenshot_2019-05-02 WordPress Themes — WordPress com

The changes in this PR look like this:

Screenshot_2019-05-02 WordPress Themes ‹ Chris' Awesomesauce Sandbox (Site Title) — WordPress com FIXED green

#### Changes proposed in this Pull Request

* Re-arrange the theme banner elements so the close button isn't being overridden by the link element. Fixes #32739
* CSS changes to accommodate the markup changes
* Fixed a `z-index` issue where the image sat above the banner copy
* Also made the banner work responsively, so the text never sits on top of the image, improving readability.

#### Testing instructions

* Browse to Design > Themes
* Note that the banner text doesn't sit on top of the theme screenshot
* Check that the X button closes the banner

### CODE REVIEW

File: client/my-sites/themes/themes-banner/style.scss [hunk 1]
```diff
@@ -79,8 +81,10 @@
 		right: 0;
 		top: 0;
 		width: 1.7em;
+		z-index: 1;
 
 		&:hover {
+			background: rgba( white, 0.8 );
```
Comment (MichaelArestad) [inline client/my-sites/themes/themes-banner/style.scss]: What's going on here?

File: client/my-sites/themes/themes-banner/style.scss [hunk 2]
```diff
@@ -25,11 +18,22 @@
 		width: 100%;
 	}
 
+	a {
+		color: white !important;
```
Comment (MichaelArestad) [inline client/my-sites/themes/themes-banner/style.scss]: These importants are killing me. I know you moved them, but maybe we should remove them. Future cleanup PR?

File: client/my-sites/themes/themes-banner/style.scss (hunk 1 above)
Comment (crunnells) [reply client/my-sites/themes/themes-banner/style.scss]: I don't know but they are _all over_ that file. I've removed all references to `white` and `black` and converted them to `rgba()` or hexcode where relevant.

File: client/my-sites/themes/themes-banner/style.scss (hunk 2 above)
Comment (crunnells) [reply client/my-sites/themes/themes-banner/style.scss]: Removed! I'm not sure _why_ they were there, as the specificity is pretty good. We'll have to keep an eye out.

**Comment by crunnells** [comment]
I should note that the last commit 6a3451b includes some tweaks to the `.themes-banner` CSS that makes the `<a>` cover the whole space. In my previous commits it only worked on the elements within the `<a>`, but not on the empty spaces. This makes the banner a little bit more clickable and effective.

**Comment by crunnells** [comment]
cc: @mendezcode and @mapk since you guys worked on the original banner

**Comment by arunsathiya** [comment]
A quick note that there is an approved PR from a contributor https://github.com/Automattic/wp-calypso/pull/32744 for fixing the theme banner close issue. I spoke with @crunnells and we decided to merge both the PRs together.

**Comment by alaczek** [comment]
> So should PR #32744 be just closed then?

Yep, I'll be closing that one. 

@crunnells adding z-index requires some extra work - see https://wpcalypso.wordpress.com/devdocs/docs/coding-guidelines/css.md#z-index
Basically you should do something similar as in https://github.com/Automattic/wp-calypso/pull/32744/files

**Pull request merged**